docs/multicluster/architecture.md

```diff
@@ -0,0 +1,95 @@
+can also be configured as a dedicated leader cluster of multiple ClusterSets.
+
+## Terminology
+
+ClusterSet is a placeholder name for a group of clusters with a high degree of mutual
+trust and shared ownership that share Services amongst themselves. Within a ClusterSet,
+Namespace sameness applies, which means all Namespaces with a given name are considered to
+be the same Namespace. It's also a Custom Resource Definition which defines a ClusterSet
+including the leader and member clusters information.
+
+ClusterClaim is a Custom Resource Definition which can be used to define a cluster itself as
+a member of a ClusterSet with unique cluster ID, or claim a ClusterSet with unique ClusterSet
+ID.
+
+MemberClusterAnnounce is a Custom Resource Definition which declares a member cluster
+configuration to the leader cluster.
+
+Common Area is an abstraction in Multi-cluster implementation that provides interfaces to a
+persistent storage that can be read/write and shared for all member clusters and leader clusters.
+
+## Antrea Multi-cluster Controller
+
+In a member cluster, Antrea Multi-cluster creates a Deployment that runs Antrea Multi-cluster
+Controller which is responsible for exporting resource to and importing resource from a leader
+cluster in a ClusterSet.
+
+In a leader cluster, Antrea Multi-cluster creates a Deployment that runs Antrea Multi-cluster
+Controller which is responsible for converting resources from different member clusters into one
+encapsulated resource as long as these resources have the same kind and match Namespace sameness.
+
+In ClusterSet initialization, Antrea Multi-cluster Controller in a member cluster watches
+ClusterSet, ClusterClaim, and creates a MemberClusterAnnounce in the leader cluster.
+
+For resource exporting, it watches ServiceExport, Service and Endpoints resources from Kubernetes
+API, encapsulates Service and Endpoints into ResourceExport according to ServiceExport, and
+write the ResourceExport to leader cluster. For resource importing, it watches ResourceImport
+from leader cluster, and creates multi-cluster Service and Endpoints with a prefix `antrea-mc-`
+plus exported Service name, and also a ServiceImport which has the same name as the exported Service.
+
+In a leader cluster, For ClusterSet initialization, Antrea Multi-cluster controller watches
+ClusterSet, ClusterClaim, and validate the ClusterSet and Clusterclaim. For resource exchange,
+it watches ResourceExport, compute ResourceExport and encapsulates them into ResourceImport.
+
+## Service Export and Import
+
+<img src="assets/resource-export-import-pipeline.png" width="800" alt="Antrea Multi-cluster Resource Export/Import Pipeline">
+
+The above diagram depicts Antrea Multi-cluster resource export/import pipeline. In the current
+implementation, it focuses on Service discovery and exchange Service resource in different
+member clusters.
+
+Below is a sample workflow about how a Kubernetes Service being exchanged within a ClusterSet.
```

Comment:
Seems this paragraph talks about the same as the next. Should we remove it?

Reply:
sorry, I didn't get this comment, remove which paragraph?